sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1451,109 +1451,6 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     })
   }
 
-  test("backward compatibility") {
-    withSQLConf(SQLConf.LEGACY_TIME_PARSER_ENABLED.key -> "true") {
-      // This test we make sure our JSON support can read JSON data generated by previous version
-      // of Spark generated through toJSON method and JSON data source.
-      // The data is generated by the following program.
-      // Here are a few notes:
-      //  - Spark 1.5.0 cannot save timestamp data. So, we manually added timestamp field (col13)
-      //      in the JSON object.
-      //  - For Spark before 1.5.1, we do not generate UDTs. So, we manually added the UDT value to
-      //      JSON objects generated by those Spark versions (col17).
-      //  - If the type is NullType, we do not write data out.
```

Comment:
What is the reason we removed this test case? This test case sounds very critical. 

cc @srowen @marmbrus @MaxGekk @cloud-fan

Reply:
This was added in the PR https://github.com/apache/spark/pull/8806/

Reply:
We have to capture all such changes before the final release of Spark 3.0. We can delay the release but we have to capture all such changes. Please let us know if you are aware of any similar change we made in the upcoming 3.0 release.

Reply:
I think we need to add more such test cases for ensuring we do not break the backward compatibility for all the built-in data sources.

Reply:
The test is gone because the old behavior is gone; that's all that's going on here.
See the OP with a link to the actual change. The key discussions were:

https://github.com/apache/spark/pull/23391#discussion_r244414750
https://github.com/apache/spark/pull/23391#discussion_r244627486

I think the TL;DR is that the legacy behavior is error-prone and already susceptible to getting wrong answers for old dates. That seems worth 'fixing' despite the forced behavior change.

Reply:
Based on the latest discussions in https://github.com/apache/spark/pull/27710#discussion_r384584040, we can't silently return the wrong results. The backward compatibility is very critical.

Reply:
I defer to @MaxGekk and @cloud-fan on that thread. It's trading one set of problems for another but it could be the right thing. We will never get rid of the legacy behavior now, I'm pretty sure :)